PokeDemo/Controllers/PokemonListViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ViewController.swift
+//  PokeDemo
+//
+//  Created by Rudr Bansal on 05/07/22.
+//
+// MARK: - Navigation
+
+import UIKit
+
+class PokemonListViewController: BaseViewController {
+    
+    // MARK: - IBOutlets
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    // MARK: - Properties
+    
+    // MARK: - Private
+    
+    private let presenter = PokemonListPresenter()
+    private var pokemons: [Pokemon]? = [Pokemon]()
+    
+    // MARK: - View Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.register(UINib(nibName: PokemonListCell.cellIdentifier(), bundle: nil), forCellReuseIdentifier: PokemonListCell.cellIdentifier())
+        tableView.dataSource = self
+        tableView.delegate = self
```

Comment:
**suggestion:** Instead of dumping all sorts of configurations in `viewDidLoad`, you can leverage the property observers -  [Refer here](https://docs.swift.org/swift-book/LanguageGuide/Properties.html)

```
@IBOutlet weak var tableView: UITableView! {
    didSet {
        tableView.register(UINib(nibName: PokemonListCell.cellIdentifier(), bundle: nil), forCellReuseIdentifier: PokemonListCell.cellIdentifier())
        tableView.dataSource = self
        tableView.delegate = self
 } 
}
```

Reply:
Covered in 65ba611f7b38b56cf726f9afed6a8efc55e5e8b2